bindings/go/src/fdb/tuple/tuple.go

```diff
@@ -314,11 +369,50 @@ func (p *packer) encodeTuple(t Tuple, nested bool) {
 // call Pack when using a Tuple with a FoundationDB API function that requires a
 // key.
 func (t Tuple) Pack() []byte {
-	p := packer{buf: make([]byte, 0, 64)}
+	p := newPacker()
 	p.encodeTuple(t, false)
 	return p.buf
 }
 
+// PackWithVersionstamp packs the specified tuple into a key for versionstamp operations
+func (t Tuple) PackWithVersionstamp() ([]byte, error) {
+	hasVersionstamp, err := t.HasIncompleteVersionstamp()
+	if err != nil {
+		return nil, err
+	}
+
+	p := newPacker()
+	p.encodeTuple(t, false)
+
+	if hasVersionstamp {
+		var scratch [4]byte
+		binary.LittleEndian.PutUint32(scratch[:], uint32(p.versionstampPos))
+		p.putBytes(scratch[:])
+	}
+
+	return p.buf, nil
+}
+
+// HasIncompleteVersionstamp determines if there is at least one incomplete versionstamp in a tuple
+func (t Tuple) HasIncompleteVersionstamp() (bool, error) {
```

Comment:
Hm, I'm not necessarily opposed to having this return an `error` if the number of versionstamps is greater than 1 as this means that, like, `PackWithVersionstamp` can return an `error` without needing to recover (which is painful). (Though we can eschew the need to `recover` if `encodeTuple` can return an `error`.) The other bindings, for what it's worth, will not throw an error in that case.

Reply:
https://github.com/apple/foundationdb/blob/af69ba035a8c9d9ff3d15a5d40e716ecca373a57/bindings/python/fdb/tuple.py#L384

This from the python bindings seems conceptually similar to what I did here. I just extracted `HasIncompleteVersionstamp` into a function instead of it being a part of something like `_pack_maybe_with_versionstamp` in the python bindings.